govcd/openapi_endpoints_unit_test.go

```diff
@@ -34,40 +44,60 @@ func TestClient_getOpenApiHighestElevatedVersion(t *testing.T) {
 			wantErr:           true, // NAT requires at least version 34.0
 		},
 		{
-			name:              "VCD_minimum_required_version_only",
-			supportedVersions: renderSupportedVersions([]string{"28.0", "29.0", "30.0", "31.0", "32.0", "33.0", "34.0"}),
-			endpoint:          types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointNsxtNatRules,
-			wantVersion:       "34.0",
-			wantErr:           false, // NAT minimum requirement is version 34.0
+			name:                        "VCD_minimum_required_version_only",
+			supportedVersions:           renderSupportedVersions([]string{"28.0", "29.0", "30.0", "31.0", "32.0", "33.0", "34.0"}),
+			endpoint:                    types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointNsxtNatRules,
+			wantVersion:                 "34.0",
+			wantErr:                     false, // NAT minimum requirement is version 34.0
+			overRideclientMinApiVersion: "34.0",
 		},
 		{
 			name: "VCD_minimum_required_version_only_unordered",
 			// Explicitly pass in unordered VCD API supported versions to ensure that ordering and matching works well
-			supportedVersions: renderSupportedVersions([]string{"33.0", "34.0", "30.0", "31.0", "32.0", "28.0", "29.0"}),
-			endpoint:          types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointNsxtNatRules,
-			wantVersion:       "34.0",
-			wantErr:           false, // NAT minimum requirement is version 34.0
+			supportedVersions:           renderSupportedVersions([]string{"33.0", "34.0", "30.0", "31.0", "32.0", "28.0", "29.0"}),
+			endpoint:                    types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointNsxtNatRules,
+			wantVersion:                 "34.0",
+			wantErr:                     false, // NAT minimum requirement is version 34.0
+			overRideclientMinApiVersion: "34.0",
+		},
+		{
+			name:              "VCD_version_higher_than_elevated_version_entries",
+			supportedVersions: renderSupportedVersions([]string{"37.0", "37.1", "37.2", "37.3", "38.0", "38.1", "39.0"}),
+			endpoint:          types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointFirewallGroups,
+			wantVersion:       minVcdApiVersion,
+			wantErr:           false,
```

Comment:
An exact case that replicate the bug - `types.OpenApiEndpointFirewallGroups` only has one elevation directive in `endpointElevatedApiVersions`:

```go
types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointFirewallGroups: {
		//"34.0", // Basic minimum required version
		"36.0", // Adds support for Dynamic Security Groups by deprecating `Type` field in favor of `TypeValue`
	},
```